src/System.Text.Json/src/System/Text/Json/Utf8JsonReader.MultiSegment.cs

```diff
@@ -646,21 +646,18 @@ private bool ConsumeNumberMultiSegment()
 
             if (_consumed >= (uint)_buffer.Length)
             {
+                Debug.Assert(IsLastSpan);
+
                 if (!_isNotPrimitive)
```

Comment:
Can you remove this as well and replace with `Debug.Assert(_isNotPrimitive);`? And if not, we are missing a test case since code coverage never hits the `return true` path.

Reply:
If you would like to see the code coverage numbers (which might be a bit out-dated), I added them to this issue: https://github.com/dotnet/corefx/issues/34570

Locally, you can run the following from the test directory: `dotnet msbuild /t:BuildAndTest /p:Coverage=true`

https://github.com/dotnet/corefx/blob/master/Documentation/building/code-coverage.md

Reply:
> How shall we proceed?

Remove the if check and resolve the merge conflicts.

> Can you shed some light on the thoughts behind the `if` ?

We have `if (!_isNotPrimitive)` checks else where to throw when there is more data after a single JSON value. That is the main intent being the field.

I believe I had added it to ConsumeNumber as part of the initial implementation where ConsumeNumber was being called for reading the first value as well (from ReadFirstToken), so they  check was necessary at the time to make sure that there is nothing else after the single JSON number.

If you notice, we don't call ConsumeNumber from ReadFirstToken anymore, and have a special case for it (with checks for end of data after). Therefore, it is essentially leftover code that I forgot to remove as part of the update.

https://github.com/dotnet/corefx/blob/922fe872bd8c608ce590c9692b7666de59279659/src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs#L417-L426


I would be fine with it if you don't think we should add the Debug.Assert.